integration_tests/src/main/python/map_test.py

```diff
@@ -32,6 +38,44 @@ def test_simple_get_map_value(data_gen):
                 'a["NOT_FOUND"]',
                 'a["key_5"]'))
 
+def test_map_expr_project_string_to_string():
+    data_gen = [('a', StringGen(nullable=False)), ('b', StringGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+def test_map_expr_project_int_to_int():
+    data_gen = [('a', IntegerGen(nullable=False)), ('b', IntegerGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m2"))
+
+def test_map_expr_project_string_to_int():
+    data_gen = [('a', StringGen(nullable=False)), ('b', IntegerGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                "map(a, b) as m1"))
+
+def test_map_expr_project_int_to_string():
```

Comment:
nit: I would prefer to see this parametrize. That way we can add more types simply.

```

@pytest.mark.parametrize('key_type', [StringGen, IntegerGen], ids=idfn)
@pytest.mark.parametrize('value_type', [StringGen, IntegerGen], ids=idfn)
def test_single_entry_map(key_type, value_type):
    data_gen = [('a', key_gen(nullable=False)), ('b', value_gen(nullable=False))]
    assert_gpu_and_cpu_are_equal_collect(
            lambda spark : gen_df(spark, data_gen).selectExpr(
                "map(a, b) as m1"))
```